### PR TITLE
[ #1428] Support ZuulBlockFallbackProvider to return information custom

### LINKED ISSUE #1428: 网关限流异常类ZuulBlockFallbackProvider中返回信息限定了BlockResponse，能不能支持自定义
## Issue Description

Type: *bug report* or *feature request*

### How to reproduce it (as minimally and precisely as possible)

1. 
2. 
3.

### PR DESCRIPTION
**Pull request by liuweiGit**
### Describe what this PR does / why we need it

Still returns BlockResponse but the user is customizing
ZuulBlockFallbackProvider can add other parameters in addition to the basic parameters

### Does this pull request fix one issue?

Fixes #1428
### Describe how you did it
BlockResponse response = new BlockResponse(429,"Sentinel block exception",route);
response.put("data",new ArrayList(){{}});

### CODE REVIEW

File: sentinel-adapter/sentinel-zuul-adapter/pom.xml [hunk 1]
```diff
@@ -51,14 +51,23 @@
             <version>4.3.20.RELEASE</version>
             <scope>provided</scope>
         </dependency>
-
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
```
Comment (sczyh30) [inline sentinel-adapter/sentinel-zuul-adapter/pom.xml:56]: Is it required?

File: sentinel-adapter/sentinel-zuul-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/zuul/fallback/BlockResponse.java [hunk 2]
```diff
@@ -61,12 +69,14 @@ public void setRoute(String route) {
...
+    }
+
     @Override
     public String toString() {
-        return "{" +
-                "\"code\":" + code +
-                ", \"message\":" + "\"" + message + "\"" +
-                ", \"route\":" + "\"" + route + "\"" +
-                '}';
+      return  JSON.toJSONString(this);
```
Comment (sczyh30) [inline sentinel-adapter/sentinel-zuul-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/zuul/fallback/BlockResponse.java:80]: It's not recommended to introduce JSON libraries directly here.

File: sentinel-adapter/sentinel-zuul-adapter/pom.xml (hunk 1 above)
Comment (liuweiGit) [reply sentinel-adapter/sentinel-zuul-adapter/pom.xml:56]: This is no longer needed

File: sentinel-adapter/sentinel-zuul-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/zuul/fallback/BlockResponse.java (hunk 2 above)
Comment (liuweiGit) [reply sentinel-adapter/sentinel-zuul-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/zuul/fallback/BlockResponse.java:80]: Okay, I modify it

Comment (liuweiGit) [reply sentinel-adapter/sentinel-zuul-adapter/src/main/java/com/alibaba/csp/sentinel/adapter/gateway/zuul/fallback/BlockResponse.java:80]: > It's not recommended to introduce JSON libraries directly here.

If you don't use JSON, how to parse the user-defined complex response content? Do you have a better way to deal with it?

**Pull request closed without merging**